src/components/vm/hostdevs/hostDevCard.jsx

```diff
@@ -0,0 +1,215 @@
+ *
+ * Copyright (C) 2021 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import cockpit from 'cockpit';
+import { vmId, findHostNodeDevice } from "../../../helpers.js";
+import { ListingTable } from "cockpit-components-table.jsx";
+
+const _ = cockpit.gettext;
+
+/* Adds an optional description-value pair to an array which represents multiple values of a table cell
+ *
+ * @param chunks an array into which element is added to
+ * @param value a value of the descr-value pair
+ * @param descr a description of the descr-value pair
+ * @param id
+ */
+function addOptionalToCell(chunks, value, id, descr) {
+    if (value) {
+        chunks.push(
+            <div className='machines-multivalue-column' key={descr}>
+                <strong className='machines-multivalue-column-label'>
+                    {descr}
+                </strong>
+                <span className='machines-multivalue-column-value' id={id}>
+                    {value}
+                </span>
+            </div>
+        );
+    }
+}
+
+addOptionalToCell.propTypes = {
+    chunks: PropTypes.array.isRequired,
+    value: PropTypes.oneOf(['string', 'number']),
+    descr: PropTypes.oneOf(['string', 'number']),
+    id: PropTypes.string.isRequired,
+};
+
+export const VmHostDevCard = ({ vm, nodeDevices, config, onAddErrorNotification }) => {
+    function getClass(hostDev, hostdevId) {
+        const nodeDev = findHostNodeDevice(hostDev, nodeDevices);
+
+        if (nodeDev && (["usb_device", "pci"].includes(nodeDev.capability.type)))
+            return nodeDev.class;
+    }
+
+    function getProduct(hostDev, hostdevId) {
+        const nodeDev = findHostNodeDevice(hostDev, nodeDevices);
+
+        if (nodeDev && (["usb_device", "pci"].includes(nodeDev.capability.type)))
+            return nodeDev.capability.product._value;
+    }
+
+    function getVendor(hostDev, hostdevId) {
+        const nodeDev = findHostNodeDevice(hostDev, nodeDevices);
+
+        if (nodeDev && (["usb_device", "pci"].includes(nodeDev.capability.type)))
+            return nodeDev.capability.vendor._value;
+    }
+
+    function getSource(hostDev, hostdevId) {
+        const cell = [];
+        if (hostDev.type === "usb") {
+            const nodeDev = findHostNodeDevice(hostDev, nodeDevices);
```

Comment:
Please add the if nodeDev check here as well. See the example I posted which crashes it.